tests/rptest/tests/partition_state_api_test.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2022 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from rptest.services.admin import Admin
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.cluster import cluster
+from rptest.clients.types import TopicSpec
+
+
+class PartitionStateAPItest(RedpandaTest):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs, num_brokers=5)
+
+    def _validate_states(self, states, num_replicas, has_leader=True):
```

Comment:
do you think it is worth sanity checking more of the result, or even checking the schema? might be overkill.

Reply:
May be an overkill, this is a v0 of a debug endpoint, pretty sure this will change as we add more fields as needed for debugging incidents. (eg: stm specific items in the state)